test/unit/generator/content/schedule/part3/get-top-up-summary.test.js

```diff
@@ -0,0 +1,31 @@
+const toCurrencyString = require('../../../../../../app/generator/to-currency-string')
+const getTopUpSummary = require('../../../../../../app/generator/content/schedule/part3/get-top-up-summary')
+const { topUpAdjustment } = require('../../../../../mocks/objects/adjustment')
+
+describe('Get top-up summary', () => {
+  test('should return "The table below shows your revised payments." for key stack 6 entry', () => {
+    const result = getTopUpSummary(topUpAdjustment)
+    expect(result.stack[0]).toBe('The table below shows your revised payments.\n\n')
+  })
+
+  test('should use solid line style for grid', () => {
+    const result = getTopUpSummary(topUpAdjustment)
+    expect(result.stack[1].layout.hLineStyle()).toBe('solid')
+    expect(result.stack[1].layout.vLineStyle()).toBe('solid')
+  })
+
+  test('should return "Current agreement value: £1,000.00\n" for key stack 2 line 1` entry', () => {

```

Comment:
Again, would consider changing the test name to reflect `topUpAdjustment.currentValue`